homeassistant/components/zha/api.py

```diff
@@ -256,11 +147,11 @@ def async_load_api(hass, application_controller, listener):
         attribute = service.data.get(ATTR_ATTRIBUTE)
         value = service.data.get(ATTR_VALUE)
         manufacturer = service.data.get(ATTR_MANUFACTURER) or None
-        component = hass.data.get(entity_id.split('.')[0])
-        entity = component.get_entity(entity_id)
+        entity = zha_gateway.get_entity(entity_id)
```

Comment:
As noted below, we shouldn't access the entity directly from outside its platform. Use our dispatch helper instead to signal the entity if needed. If the only the `zha_device` instance needs access, maybe we can access that not via the entity some way?